tox_pypi_filter/plugin.py

```diff
@@ -22,54 +20,41 @@
 
 HELP = ("Specify version constraints for packages which are then applied by "
         "setting up a proxy PyPI server. If giving multiple constraints, you "
-        "can separate them with semicolons (;).")
-
-HELP_REQ = ("Specify version constraints for packages which are then applied by "
-            "setting up a proxy PyPI server. This should be set to a file in "
-            "pip requirements.txt format, i.e. one package per line, and can "
-            "be a URL.")
+        "can separate them with semicolons (;). This can also be a URL to a "
+        "local file (e.g. file:// or file:///) or a remote file (e.g. "
+        "http://, https://, or ftp://).")
 
 
 @hookimpl
 def tox_addoption(parser):
-
     parser.add_argument('--pypi-filter', dest='pypi_filter', help=HELP)
-    parser.add_argument('--pypi-filter-requirements', dest='pypi_filter_req',
-                        help=HELP_REQ)
-
     parser.add_testenv_attribute('pypi_filter', 'string', help=HELP)
-    parser.add_testenv_attribute('pypi_filter_requirements', 'string',
-                                 help=HELP_REQ)
 
 
 SERVER_PROCESS = []
 
+
 @hookimpl
 def tox_testenv_create(venv, action):
     global SERVER_PROCESS
 
     pypi_filter = venv.envconfig.config.option.pypi_filter or venv.envconfig.pypi_filter
-    pypi_filter_req = venv.envconfig.config.option.pypi_filter_req or venv.envconfig.pypi_filter_requirements
 
-    if not pypi_filter and not pypi_filter_req:
+    if not pypi_filter:
         return
 
-    if pypi_filter and pypi_filter_req:
-        raise ValueError("Please specify only one of --pypi-filter or --pypi-filter-requirements")
+    url_info = urllib.parse.urlparse(pypi_filter)
 
-    if pypi_filter:
+    if url_info.scheme == 'file':
+        reqfile = url_info.netloc
```

Comment:
I don't think this works if you have a absolute file path. I think you need this:
```suggestion
        reqfile = url_info.netloc + url_info.path
```